src/PhpSpreadsheet/Writer/Xlsx/StringTable.php

```diff
@@ -42,35 +29,33 @@ public function createStringTable(Worksheet $pSheet, $pExistingTable = null)
                 !is_object($cellValue) &&
                 ($cellValue !== null) &&
                 $cellValue !== '' &&
-                ($cell->getDataType() == DataType::TYPE_STRING || $cell->getDataType() == DataType::TYPE_STRING2 || $cell->getDataType() == DataType::TYPE_NULL) &&
-                !isset($aFlippedStringTable[$cellValue])
+                !$aFlippedStringTable->has(md5($cellValue)) &&
+                ($cell->getDataType() == DataType::TYPE_STRING || $cell->getDataType() == DataType::TYPE_STRING2 || $cell->getDataType() == DataType::TYPE_NULL)
             ) {
-                $aStringTable[] = $cellValue;
-                $aFlippedStringTable[$cellValue] = true;
+                fwrite($stringTableFD, serialize($cellValue) . PHP_EOL);
```

Comment:
The second issue I found is that serialize of a string with a newline does **not** encode the newline char at all, so it will clash with the `PHP_EOL` and will fail during the `writeStringTable` function below when running fgets (only get up to the internal newline and unserialize will fail).

I added in my branch a use of `base64_encode` (and decode) to work around this issue.